packages/network-controller/tests/NetworkController.test.ts

```diff
@@ -5445,7 +5576,7 @@ describe('NetworkController', () => {
 
             await waitForStateChanges(messenger, {
               propertyPath: ['networkId'],
-              count: 1,
+              count: 2,
```

Comment:
It seems that the network ID we're checking for here was _always_ the second state change. But the second state change was made synchronously after the first one, so the test still passed. The addition of the second async call spoiled this timing coincidence.